packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ComposeViewProp.kt

```diff
@@ -0,0 +1,31 @@
+package expo.modules.kotlin.views
+
+import android.view.View
+import androidx.compose.runtime.MutableState
+import com.facebook.react.bridge.Dynamic
+import expo.modules.kotlin.AppContext
+import expo.modules.kotlin.exception.PropSetException
+import expo.modules.kotlin.exception.exceptionDecorator
+import expo.modules.kotlin.types.AnyType
+import kotlin.reflect.full.memberProperties
+
+class ComposeViewProp(
+  name: String,
+  anyType: AnyType
+) : AnyViewProp(name, anyType) {
+
+  @Suppress("UNCHECKED_CAST")
+  override fun set(prop: Dynamic, onView: View, appContext: AppContext?) {
+    exceptionDecorator({
+      PropSetException(name, onView::class, it)
+    }) {
+      val props = (onView as ExpoComposeView).props ?: return
+      val property = props::class.memberProperties.find { it.name == name }?.getter?.call(onView.props)
```

Comment:
do we have any better ways than the `memberProperties` reflection here?

i have a feeling that the `ExpoComposeView.props` has duplicated work with expo-modules-core's props. for example, if we can have dsl like this, we don't necessarily have `ExpoComposeView.props`:
```kotlin
View(SingleChoiceSegmentedControlView::class) {
  Prop("options") { view: SingleChoiceSegmentedControlView, options: Array<String> ->
    (view.options as MutableState<Any?>).value = options
  }
}

class SingleChoiceSegmentedControlView(context: Context, appContext: AppContext) : ExpoComposeView(context, appContext) {
  var options by remember { mutableStateOf(emptyArray()) }
  // ...  
}
```

just my two cents here. i'm leaving the final confirmation for @lukmccall. he should have better insight than me.

Reply:
Yeah, we could always only rely on prop setters in existing DSL and not have any special handling for compose props – your code probably already works out of the box, but i felt needing to define setters was repeatable and boilerplate'y – I'll defer to @lukmccall for his opinion :)

Reply:
knowing that we have `ExpoSwiftUI.ViewProps` on ios already. maybe it also makes sense to align the dsl. i'm approving the pr anyway